File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
// ...

namespace PassParametersSilently {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            // Create a report instance.
            XtraReport1 report = new XtraReport1();

            // Obtain a parameter, and set its value.
            report.catID.Value = 2;

            // Hide the Parameters UI from end-users.
            report.catID.Visible = false;

            // Show the report's print preview.
            ReportPrintTool printTool = new ReportPrintTool(report);
            printTool.ShowPreviewDialog();
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Windows.Forms
' ...

Namespace PassParametersSilently
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) _
		Handles button1.Click
			' Create a report instance.
			Dim report As New XtraReport1()

			' Set the parameter's value.
			report.catID.Value = 3

			' Hide the Parameters Request pane by default in preview.
			report.RequestParameters = False

			' Show the report's print preview.
			report.ShowPreview()
		End Sub
	End Class
End Namespace